examples/TaskLoop/main.roc

```diff
@@ -0,0 +1,60 @@
+app "task-usage"
+    packages {
+        pf: "https://github.com/roc-lang/basic-cli/releases/download/0.7.1/Icc3xJoIixF3hCcfXrDwLCu4wQHtNdPyoJkEbkgIElA.tar.br",
+    }
+    imports [pf.Stdin, pf.Stdout, pf.Stderr, pf.Task.{ Task }]
+    provides [main] to pf
+
+main =
+    run |> Task.onErr handleErr
+
+run =
+    # Start the loop of tasks defined by `addNumberFromStdinT`. The initial state, the total, is 0.
+    # `total` will be set with the final step of the loop, when `addNumberFromStdinT` returns `Task.ok (Done total)`
+    total <- Task.loop 0 addNumberFromStdinT |> Task.await
+    Stdout.line "Total: \(Num.toStr total)"
+
+# This function read from stdin and return one of the following Tasks:
+# `Task.ok (Step aNewState)`: To inform Task.loop that a new Step has been completed with a new state.
+# `Task.ok (Done aFinalState)`: To inform Task.loop that the loop is done with a final state. No new steps will be executed.
+# `Task.err error`: To inform Task.loop that an error has occurred, in this case an unprocessable input, and the loop should be stopped.
+addNumberFromStdinT = \total ->
+    # Read a line from stdin
+    line <- Stdin.line |> Task.await
+    # Use addNumberFromStdin to calculate the step
+    when addNumberFromStdin total line is
+        # The step has been completed or no further steps to execute
+        Ok stepOrDone -> Task.ok stepOrDone
+        # The inpput cannot be processed. `Task.loop` will return with error
+        Err err -> Task.err err
+
+# This function takes the current state, the total, and a line from stdin and return one of the three possible values
+addNumberFromStdin = \total, line ->
```

Comment:
I decided to split the logic between `addNumberFromStdinT` and `addNumberFromStdin` functions, with the second to be "Task-free".

This allowed me to test the latter as a pure function. I know `addNumberFromStdinT` is pure as well as it returns just `Task` data, but, to be honest, I don't know how to test it by mocking `Stdin.line` :thinking: .

Is this an acceptable and idiomatic approach or can be done better?

Reply:
@Anton-4 , this is ready for review, if you have time, please. However, I'm not completely sure about my approach. See prev. comment.